pkg/backporter/backporter.go

```diff
@@ -553,30 +574,111 @@ func CreateCloneHandler(client bugzilla.Client, allTargetVersions sets.String, m
+			handleError(w, err, fmt.Sprintf("unable to convert \"ID\" parameter from string to int: %s", req.FormValue("ID")), http.StatusBadRequest, req.URL.Path, 0, m)
+			return
+		}
+		sortByTargetRelease(clones)
+		releaseInvalidErrorMsg := "%s cannot be parsed as release, must be of the form X.X.X"
+		toCloneMajorMinorRelease, err := getMajorMinorRelease(req.FormValue("release"))
 		if err != nil {
-			handleError(w, err, "clone creation failed", http.StatusInternalServerError, endpoint, bugID, m)
+			handleError(w, err, fmt.Sprintf(releaseInvalidErrorMsg, req.FormValue("release")), http.StatusBadRequest, endpoint, bugID, m)
 			return
 		}
-		targetRelease := bugzilla.BugUpdate{
-			TargetRelease: []string{
-				req.FormValue("release"),
-			},
+		var sourceBug *bugzilla.Bug
+		var sourceBugMajorMinorRel string
+		for _, clone := range clones {
+			if !isTargetReleaseSet(clone) {
+				continue
+			}
+			cloneMajorMinorRel, err := getMajorMinorRelease(clone.TargetRelease[0])
+			if err != nil {
+				handleError(w, err, fmt.Sprintf(releaseInvalidErrorMsg, clone.TargetRelease[0]), http.StatusBadRequest, endpoint, bugID, m)
+				return
+			}
+			if toCloneMajorMinorRelease == cloneMajorMinorRel {
+				handleError(w, err, fmt.Sprintf("clone for major release %s already exists", clone.TargetRelease[0]), http.StatusBadRequest, endpoint, bugID, m)
+				return
+			}
+			if cloneMajorMinorRel > toCloneMajorMinorRelease {
+				sourceBug = clone
+				sourceBugMajorMinorRel = cloneMajorMinorRel
+				break
+			}
 		}
-		// Updating the cloned bug with the right target version
-		if err := client.UpdateBug(cloneID, targetRelease); err != nil {
-			handleError(w, err, fmt.Sprintf("clone created - Bug#%d, but failed to specify version for the cloned bug.", cloneID), http.StatusInternalServerError, endpoint, bugID, m)
+		if sourceBug == nil {
+			handleError(w, err, fmt.Sprintf("one bug with greater release needs to be present to clone from (highest release present is %s)", clones[len(clones)-1].TargetRelease[0]), http.StatusBadRequest, endpoint, bugID, m)
 			return
 		}
+		var sortedMajorRelease []string
+		for release := range allTargetVersions {
+			mRel, err := getMajorMinorRelease(release)
+			if err != nil {
+				handleError(w, err, fmt.Sprintf(releaseInvalidErrorMsg, release), http.StatusInternalServerError, endpoint, bugID, m)
+				return
+			}
+			sortedMajorRelease = append(sortedMajorRelease, mRel)
+		}
+		// Sort the releases in descending order
+		sort.SliceStable(sortedMajorRelease, func(i, j int) bool {
+			return sortedMajorRelease[i] > sortedMajorRelease[j]
+		})
+		var targetRelease int
+		for i, rel := range sortedMajorRelease {
+			if rel == sourceBugMajorMinorRel {
+				targetRelease = i + 1
+				break
+			}
+		}
+		var newClones []string
+		finalTargetFound := false
+		// Find source bug and keep iterating till we hit the target release
+		for !finalTargetFound {
+			if sortedMajorRelease[targetRelease] == toCloneMajorMinorRelease {
```

Comment:
There should technically never be that scenario as the sourceBugMajorMinorRel must be greater than at least one release. But adding it would probable be a good precaution